src/main/java/com/spotify/asyncdatastoreclient/QueryBuilder.java

```diff
@@ -28,7 +28,11 @@
  * Note that it could be convenient to use an 'import static' to use the
  * methods of this class.
  */
-public class QueryBuilder {
+public final class QueryBuilder {
+
+  private QueryBuilder() throws InstantiationException {
+    throw new InstantiationException("This class is not for instantiation");
```

Comment:
I'm not sure I like throwing an exception here, it just seems unnecessary.

Declaring a `private` constructor should be enough.